docs/screen-options-resolution.md

```diff
@@ -70,14 +70,150 @@ export default function App() {
+Imagine the following configuration:
 
-## A tab navigator contains a stack and you want to hide the tab bar on specific screens
+```js
+const Tab = createBottomTabNavigator();
+
+function HomeTabs() {
+  return (
+    <Tab.Navigator>
+      <Tab.Screen name="Feed" component={FeedScreen} />
+      <Tab.Screen name="Profile" component={ProfileScreen} />
+      <Tab.Screen name="Account" component={AccountScreen} />
+    </Tab.Navigator>
+  );
+}
+
+const Stack = createStackNavigator();
+
+function App() {
+  return (
+    <Stack.Navigator>
+      <Stack.Screen name="Home" component={HomeTabs} />
+      <Stack.Screen name="Settings" component={SettingsScreen} />
+    </Stack.Navigator>
+  );
+}
+```
+
+If we were to set the `headerTitle` with `options` for the `FeedScreen`, this would not work. This is because `App` stack will only look at its immediate children for configuration: `HomeTabs` and `SettingsScreen`.
+
+But we can determine the `headerTitle` option based on the navigation state of our tab navigator using the `route.state` property. Let's create a function to get the title from `route.state` first:
+
+```js
+function getHeaderTitle(route) {
+  // Access the tab navigator's state using `route.state`
+  const routeName = route.state
+    ? // Get the currently active route name in the tab navigator
+      route.state.routes[route.state.index].name
+    : // If state doesn't exist, we need to default to the initial screen
+      // In our case, it's "Feed" as that's the first screen inside the navigator
+      'Feed';
+
+  switch (routeName) {
+    case 'Feed':
+      return 'News feed';
+    case 'Profile':
+      return 'My profile';
+    case 'Account':
+      return 'My account';
+  }
+}
+```
+
+Then we can use this function in 2 ways:
+
+1. Using `options` prop on `Screen` (recommended):
+
+   ```js
+   <Stack.Screen
+     name="Home"
+     component={HomeTabs}
+     options={({ route }) => ({
+       headerTitle: getHeaderTitle(route),
+     })}
+   />
+   ```
+
+2. Using `navigation.setOptions`:
+
+   ```js
+   function HomeTabs({ navigation, route }) {
+     navigation.setOptions({ headerTitle: getHeaderTitle(route) });
+
+     return (
+       <Tab.Navigator>
+         <Tab.Screen name="Feed" component={FeedScreen} />
+         <Tab.Screen name="Profile" component={ProfileScreen} />
+         <Tab.Screen name="Account" component={AccountScreen} />
+       </Tab.Navigator>
+     );
+   }
+   ```
+
+So what's happening here? The `route` prop contains a `state` property which refers to the child navigator's state (in this case it's the tab navigator since that's what we're rendering). We are getting the value of the currently active route name from this state and setting appropriate title for the header.
+
+> Note: The `route.state` property may not exist at all. This will always happen if we have never navigated inside the tab navigator. So it's very important to handle this case, otherwise your app will crash.
```

Comment:
```suggestion
> Note: The `route.state` property may not exist at all. This will always happen if we have never navigated inside the tab navigator. So it's very important to handle this case, otherwise, your app will crash.
```